### PR TITLE
KOGITO-1829: BDD tests: Test that Spring boot cloud events are correc…

### PR DESCRIPTION
**Pull request by Sgitario**
JIRA Ticket: https://issues.redhat.com/browse/KOGITO-1829
Description: We are currently testing only Quarkus. We should also test it works on a Spring Boot process.
I've also moved the existing test for data-index/quarkus test into the quarkus feature test as done in spring boot.

It depends on: https://github.com/kiegroup/kogito-examples/pull/216

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: test/features/deploy_quarkus_service.feature [hunk 1]
```diff
@@ -118,3 +118,39 @@ Feature: Deploy quarkus service
     Examples: Native
       | native  | minutes |
       | enabled | 20      |
+
+#####
+
+  @events
+  @persistence
+  Scenario Outline: Process instance events are stored in Data Index
```
Comment (radtriste) [inline test/features/deploy_quarkus_service.feature:126]: Can you add `Quarkus` into the title ?

File: test/features/deploy_springboot_service.feature [hunk 2]
```diff
@@ -59,3 +59,29 @@ Feature: Deploy spring boot service
     # Implement retrieving of job information from Jobs service once https://issues.redhat.com/browse/KOGITO-1163 is fixed
     Then Kogito application "process-timer-springboot" log contains text "Before timer" within 1 minutes
     And Kogito application "process-timer-springboot" log contains text "After timer" within 1 minutes
+
+#####
+
+  @events
+  @persistence
+  Scenario: Process instance events are stored in Data Index
```
Comment (radtriste) [inline test/features/deploy_springboot_service.feature:67]: Add `Spring Boot` into the title, so we can differentiate them

File: test/features/deploy_quarkus_service.feature (hunk 1 above)
Comment (Sgitario) [reply test/features/deploy_quarkus_service.feature]: Done

File: test/features/deploy_springboot_service.feature (hunk 2 above)
Comment (Sgitario) [reply test/features/deploy_springboot_service.feature]: Done

File: test/features/deploy_quarkus_service.feature [hunk 3]
```diff
@@ -118,3 +118,39 @@ Feature: Deploy quarkus service
     Examples: Native
       | native  | minutes |
       | enabled | 20      |
+
+#####
+
+  @events
+  @persistence
+  Scenario Outline: Deploy Quarkus process instance with events stored in Data Index
```
Comment (radtriste) [inline test/features/deploy_quarkus_service.feature:126]: now make me think we could improve the name :D

something like `Retrieve Quarkus Process Instance events from Data Index`

wdyt ? make sense ?
@sutaakar @MarianMacik also ?

Comment (Sgitario) [reply test/features/deploy_quarkus_service.feature:126]: Hehe, I do agree with you. What about this name: "Deploy process-quarkus-example service with cloud events" ?

Comment (radtriste) [reply test/features/deploy_quarkus_service.feature:126]: nope I think the scenario's name should show what the `Then` steps do

Comment (radtriste) [reply test/features/deploy_quarkus_service.feature:126]: and maybe the `When` if not enough explicit

Comment (Sgitario) [reply test/features/deploy_quarkus_service.feature]: As per @radtriste suggested, I changed the title to "Data Index retrieves Spring Boot process' events"

**Comment by Sgitario** [comment]
Test ran ok after rebasing and fixing conflicts.

**Comment by radtriste** [comment]
https://github.com/kiegroup/kogito-examples/pull/216 has been merged

**Pull request merged**